src/ParamHandler.ts

```diff
@@ -45,10 +47,10 @@ export class ParamHandler {
         if (param.isRequired) {
             // todo: make better error messages here
             if (param.name && isValueEmpty) {
-                return Promise.reject("Parameter " + param.name + " is required for request on " + request.method + " " + request.url);
+                return Promise.reject(new ParameterRequiredError(request.url, request.method, param.name));
 
             } else if (!param.name && (isValueEmpty || isValueEmptyObject)) {
-                return Promise.reject("Request body is required for request on " + request.method + " " + request.url);
+                return Promise.reject(new BodyRequiredError(request.url, request.method));
```

Comment:
It's weird that the `BodyRequiredError` exist in code but hasn't been used anywhere else.
Maybe @pleerock started refactoring and created it but have forgotten to change that lines 😉